compiler/planner/pattern_plan.rs

```diff
@@ -134,42 +151,93 @@ impl PatternPlan {
+                        }
+                        let planner = elements[index].as_role_player().unwrap();
+                        let selected_variables = &[rp.relation(), rp.player(), rp.role_type()];
+                        let instruction = if bound_variables.contains(&rp.relation()) {
+                            ConstraintInstruction::RolePlayer(RolePlayerInstruction::new(
+                                rp.clone(),
+                                Inputs::Single([rp.relation()]),
+                                type_annotations,
+                            ))
+                        } else if bound_variables.contains(&rp.player()) {
+                            ConstraintInstruction::RolePlayerReverse(RolePlayerReverseInstruction::new(
+                                rp.clone(),
+                                Inputs::Single([rp.player()]),
+                                type_annotations,
+                            ))
+                        } else if planner.unbound_is_forward {
+                            ConstraintInstruction::RolePlayer(RolePlayerInstruction::new(
+                                rp.clone(),
+                                Inputs::None([]),
+                                type_annotations,
+                            ))
+                        } else {
+                            ConstraintInstruction::RolePlayerReverse(RolePlayerReverseInstruction::new(
+                                rp.clone(),
+                                Inputs::None([]),
+                                type_annotations,
+                            ))
+                        };
+                        let sort_variable = if bound_variables.is_empty() && planner.unbound_is_forward
+                            || bound_variables.contains(&rp.player())
+                        {
+                            rp.relation()
+                        } else {
+                            rp.player()
+                        };
+                        steps.push(Step::Intersection(IntersectionStep::new(
+                            sort_variable,
+                            vec![instruction],
+                            selected_variables,
+                        )));
+                    }
                     Constraint::Has(has) => {
-                        let intersection_step = if bound_variables.is_empty() {
-                            IntersectionStep::new(
-                                has.owner(),
-                                vec![ConstraintInstruction::Has(HasInstruction::new(
-                                    has.clone(),
-                                    Inputs::None([]),
-                                    program.entry_annotations(),
-                                ))],
-                                &[has.owner(), has.attribute()],
-                            )
-                        } else if bound_variables.len() == 2 {
+                        if bound_variables.len() == 2 {
```

Comment:
You wanna make these branches `todo!()` so they fail if we ever hit them in future match  BDD?